tictactoe.py

```diff
@@ -1,24 +1,25 @@
 import random
 from collections import Counter
 
-player_modes = ['user', 'easy', 'medium']
-
+player_modes = ['user', 'easy', 'medium', 'unbeatable']
+INFINITY = 999999
 
 class Board:
+
     def __init__(self, config: str = ' ' * 9):
         config = config.replace('_', ' ')
         self.config = config
 
-    def get_sq(self, x, y):
-        return self.config[(3 - y) * 3 + x - 1]
+    def get_sq(self, x, y, config):
+        return config[(3 - y) * 3 + x - 1]
 
-    def whose_turn(self):
-        c = Counter(self.config)
+    def whose_turn(self, config):
```

Comment:
you could set the default parameter of the config parameter to self.config so that you need not unnecessarily pass it everytime

Reply:
I am using the same method inside the min_max method to get the turn for min/ max for each branched configs,

Reply:
Resolved here: bc6c50cd702c940c930ad708f5e11470987e4fb9